src/class.c

```diff
@@ -776,7 +777,30 @@ void ClassDeclaration::semantic(Scope *sc)
     Module::dprogress++;
 
     dtor = buildDtor(sc);
+    if (Dsymbol *assign = search_function(this, Id::assign))
+    {
+        Expression *e = new NullExp(loc, type); // dummy rvalue
+        Expressions *arguments = new Expressions();
+        arguments->push(e);
+
+        // check identity opAssign exists
+        FuncDeclaration *fd = assign->isFuncDeclaration();
+        if (fd)
+        {   fd = fd->overloadResolve(loc, e, arguments, 1);
+            if (fd && !(fd->storage_class & STCdisable))
+                goto Lassignerr;
+        }
 
+        if (TemplateDeclaration *td = assign->isTemplateDeclaration())
+        {   fd = td->deduceFunctionTemplate(sc, loc, NULL, e, arguments, 1+2);
+            if (fd && !(fd->storage_class & STCdisable))
+                goto Lassignerr;
+        }
+
+Lassignerr:
+        if (fd && !(fd->storage_class & STCdisable))
+            error("identity assignment operator overload is illegal");
```

Comment:
Because `@disable opAssign` can disallow class reference rebinding. If we allow normal opAssign for class types, it will cause null dereference if assigned class reference is null. On the other hand, disabled opAssign is always checked in compile time and no runtime error happens. I thought it would be useful.